consensus/src/experimental/buffer_item.rs

```diff
@@ -31,39 +32,79 @@ fn generate_commit_proof(
     )
 }
 
-fn aggregate_ledger_info(
+fn verify_signatures(
+    unverified_signatures: &PartialSignatures,
+    validator: &ValidatorVerifier,
+    commit_ledger_info: &LedgerInfo,
+) -> PartialSignatures {
+    // Returns a valid partial signature from a set of unverified signatures.
+    // TODO: Validating individual signatures in expensive. Replace this with optimistic signature
+    // verification for BLS.
+    PartialSignatures::new(
```

Comment:
Here, we can implement a tree-based batch verification technique that filters out invalid signature shares much faster when there are only a few of them (e.g., [LM07]: **Finding Invalid Signatures in Pairing-Based Batches**, by Law, Laurie and Matt, Brian J., *in Cryptography and Coding*, 2007).

Let me know if I should implement this.

Reply:
Added a TODO here - please feel free to take a stab at fancier approach here after we land my naive approach :)